### PR TITLE
Fix Vec support for Fuel

### PR DESCRIPTION
**Pull request by DZakh**

### CODE REVIEW

File: codegenerator/cli/templates/dynamic/codegen/src/db/Enums.res.hbs [hunk 1]
```diff
@@ -21,18 +21,11 @@ module ContractType = {
     | @as("{{contract.name.capitalized}}") {{contract.name.capitalized}}
   {{/each}}
 
-  let schema = 
-    {{#if codegen_contracts.1}}
-    S.union([
+  let schema = S.enum([
```
Comment (JonoPrest) [inline codegenerator/cli/templates/dynamic/codegen/src/db/Enums.res.hbs:24]: Is this a new feature in RescriptSchema?

File: scenarios/fuel_test/contracts/all-events/src/main.sw [hunk 2]
```diff
@@ -116,5 +116,10 @@ impl AllEvents for Contract {
 
         let data: Status = Status::Failed(StatusFailure { reason: 1 });
         log(data);
+
+        let mut vec: Vec<u64> = Vec::new();
+        vec.push(69);
```
Comment (JonoPrest) [inline scenarios/fuel_test/contracts/all-events/src/main.sw:121]: 😉

File: codegenerator/cli/src/fuel/abi.rs [hunk 3]
```diff
@@ -211,7 +211,7 @@ impl FuelAbi {
                         "b256" | "address" => String.to_ok_expr(),
                         "str" => String.to_ok_expr(),
                         type_field if type_field.starts_with("str[") => String.to_ok_expr(),
-                        "struct Vec" => Array(Box::new(GenericParam(
+                        "struct std::vec::Vec" => Array(Box::new(GenericParam(
```
Comment (JonoPrest) [inline codegenerator/cli/src/fuel/abi.rs:214]: Is this a new abi repr of a standard vec?

**Review by JonoPrest — APPROVED** [review]
Looks great Dmitry 👍🏼

File: scenarios/fuel_test/contracts/integration-tools/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+const ALL_EVENTS_CONTRACT: &str = "0x9a719b1ff9db28f1493375b06ad0195370786e217473eacc4be973ecd6281420";
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // Load .env file
+    dotenv().ok();
+
+    let phrase = env::var("MNEMONIC").expect("MNEMONIC must be set in .env");
+
+    let provider = Provider::connect("testnet.fuel.network").await.unwrap();
```
Comment (JonoPrest) [inline scenarios/fuel_test/contracts/integration-tools/src/main.rs:23]: Where does this get run btw? Or is it just for local testing?

File: codegenerator/cli/src/fuel/abi.rs (hunk 3 above)
Comment (DZakh) [reply codegenerator/cli/src/fuel/abi.rs:214]: Yep

Comment (DZakh) [reply codegenerator/cli/src/fuel/abi.rs:214]: I think it's the same, but they started prefixing struct names with modules

File: scenarios/fuel_test/contracts/integration-tools/src/main.rs (hunk 4 above)
Comment (DZakh) [reply scenarios/fuel_test/contracts/integration-tools/src/main.rs:23]: It's for local testing. I previously used the Sway Playground, but they removed the ability to call methods from there 😢

File: codegenerator/cli/templates/dynamic/codegen/src/db/Enums.res.hbs (hunk 1 above)
Comment (DZakh) [reply codegenerator/cli/templates/dynamic/codegen/src/db/Enums.res.hbs:24]: Yes

**Pull request merged**